tests/unit/test_charm_state.py

```diff
@@ -44,9 +44,8 @@ def test_charm_state_flask_config(
     harness.begin()
     harness.update_config(charm_config)
     charm_state = CharmState.from_charm(
```

Comment:
Wouldn't it be better to provide a mocked charm instead of harness here?

Reply:
Yes, it could be, but it will just be too much trouble to mock the configuration retrieving so I opt to use the harness.

Reply:
Doesn't something like the following do the trick?
```
charm = MagicMock(config=charm_config)

Reply:
I haven't thought about that, updated, thanks!

Reply:
Update other test cases where harness was only used for creating `charm_state`, thanks!